ProcessLib/Process.h

```diff
@@ -143,15 +144,21 @@ class Process
 
 private:
 	/// Creates mesh subsets, i.e. components, for given mesh.
-	void initializeMeshSubsets()
+	void initializeMeshSubsets(ProcessVariable const& variable)
 	{
 		// Create single component dof in every of the mesh's nodes.
 		_mesh_subset_all_nodes =
 		    new MeshLib::MeshSubset(_mesh, &_mesh.getNodes());
 
 		// Collect the mesh subsets in a vector.
-		_all_mesh_subsets.push_back(
-		    new MeshLib::MeshSubsets(_mesh_subset_all_nodes));
+		std::generate_n(
+		    std::back_inserter(_all_mesh_subsets),
+		    variable.getNumberOfComponents(),
+		    [&]()
+		    {
+			    return new MeshLib::MeshSubsets(_mesh_subset_all_nodes);
```

Comment:
This is a tricky one; Who has the ownership of the MeshSubsets, the Process or the DOF table?
It seems, like it should be the DOF table.